app/lib/webfinger_resource.rb

```diff
@@ -0,0 +1,60 @@
+class WebfingerResource
+  attr_reader :resource
+
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def username
+    case resource
+    when /\Ahttps?/i
+      username_from_url
+    when /acct/
```

Comment:
Thinking right now - You should be able to pass username@domain without the acct: in the front, and it should be able to deal with that. So maybe just make that the else condition.